src/neo/Network/P2P/Payloads/PluginPayloadRole.cs

```diff
@@ -0,0 +1,10 @@
+namespace Neo.Network.P2P.Payloads
+{
+    public enum PluginPayloadRole : byte
```

Comment:
I think we don't need this enum. We can create a whitelist of allowed account (`HashSet<UInt160>`). Then we can check whether the witness is included in the whitelist.

Reply:
We need to ensure that any plugin message must be signed by an official authority.

Reply:
You can cache all the official authorities into a `HashSet<UInt160>`. Then we don't need this `enum`.

Reply:
>  We can create a whitelist of allowed account (HashSet<UInt160>). Then we can check whether the witness is included in the whitelist.


I think it works.

Reply:
Done, It should be checked by the consumer

Reply:
Where is the whitelist?

Reply:
It depend to the plugin, consensus has a different one than state validators, do you want to check all together? I thought I would check it in the plugin

Reply:
The problem is relaying. Nodes that have not installed the plug-in should have a way to relay these messages.